### ISSUE TITLE
Add support for checking doctest strings in user defined functions

### ISSUE THREAD

**Issue by monga** [marimo-team/marimo#2191]
### Description

I usually put doctest strings when I define a new (non-trivial) function.

```python
def euclid_mcd(a: int, b: int) -> int:
    """Return the MCD between positive a, b.

    >>> euclid_mcd(42, 24)
    6
    >>> euclid_mcd(24, 42)
    6
    >>> euclid_mcd(42, 42)
    42
    """
    assert a > 0
    assert b > 0
    if a < b:
        a, b = b, a
    if (a != b):
        r = a - b
        return euclid_mcd(b, r)
    return a
```

I was not able to easily check the doctest in marimo. I tried the following strategies:

- `import doctest; doctest.testmodule` This normally works in a Jupyter notebook, in marimo I get "This cell raised an exception: TypeError('MarimoPdb.__init__() got an unexpected keyword argument 'nosigint'')", with Python 3.11
- `python -m doctest notebook.py` This doesn't work since the doctest string is hidden inside a `__` upper level function and ignored by `doctest`
- `marimo export script notebook.py -o notebook_test.py; python -m doctest notebook_test.py` This fails with a "ContextNotInitializedError" on a call to `mo.status.progress_bar`

Any other suggestion?

### Suggested solution

Add specific support for checking doctest strings with a cell contextual menu.

**Comment by mscolnick** [comment]
> python -m doctest notebook.py This doesn't work since the doctest string is hidden inside a __ upper level function and ignored by doctest

We plan to hoist functions out of the upper `__` - see https://github.com/marimo-team/marimo/issues/2293

> marimo export script notebook.py -o notebook_test.py; python -m doctest notebook_test.py This fails with a "ContextNotInitializedError" on a call to mo.status.progress_bar

I will look into fixing this. This is likely the best solution for now, once we fix this.

**Comment by mscolnick** [comment]
Fix for supporting for option 3: https://github.com/marimo-team/marimo/pull/2352

**Comment by mscolnick** [comment]
Also have a fix for option 1: #2353
